sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_algorithms.asciidoc

```diff
@@ -64,33 +73,42 @@ value to determine which of the extension's APIs the implementation supports.
 |===
 |Value |Description
 |1     |Initial extension version.  Base features are supported.
+|2     |Proposal is extended to other complex extensions. Multiplication binary operation is added.
 |===
 
 == Overview
 
+To reduce repetition, within this extension, `gencomplex` will be redefined to
+include `std::complex` floating point types. Therefore, `gencomplex` is defined
+as types `std::complex<float>`, `std::complex<double>`,
+`sycl::complex<sycl::half>`, `sycl::complex<float>`, and
+`sycl::complex<double>`. The extension will also refer to `mgencomplex` which
+are types `sycl::marray<sycl::complex<sycl::half>>`,
+`sycl::marray<sycl::complex<float>>`, and `sycl::marray<sycl::complex<double>>`.
+
 The types supported by some group functions and algorithms in SYCL 2020 are
 restricted to built-in scalar types and SYCL vector types. This extension
-relaxes these restrictions to permit `std::complex<float>` and
-`std::complex<double>` types.
+relaxes these restrictions to permit `gencomplex` and `mgencomplex` types.
 
 Note that the following group functions and algorithms already accept
-`std::complex<float>` arguments and `std::complex<double>` because they
-are trivially copyable:
+`gencomplex` arguments and `mgencomplex` because they are trivially copyable:
 
 - `group_broadcast`
 - `group_shift_left`
 - `group_shift_right`
 - `permute_group_by_xor`
 - `select_from_group`
 
-Usage of `std::complex<double>` requires support for double precision,
-and specifically the `sycl::aspect::fp64` device aspect.
+Usage of `gencomplex` and `mgencomplex` with double value type requires support
+for double precision, and specifically the `sycl::aspect::fp64` device aspect.
+Similarly, usage of `gencomplex` and `mgencomplex` with half value type
+requires support for half-precision, with `sycl::aspect::fp16`.
```

Comment:
Nit: This is a little confusing to me, because `gencomplex` and `mgencomplex` refer to all the value types.  "`gencomplex` with `half` value type" seems less precise than "`sycl::complex<sycl::half>` and `sycl::marray<sycl::complex<sycl::half>>>`".

Reply:
Agreed ya this is confusing. I have updated this bit to, hopefully, be much clearer.